src/linear_algebra/multilinear.lean

```diff
@@ -530,18 +530,27 @@ end semimodule
 
 section
 
-/-- Apply a permutation to the order of the arguments, obtaining another multilinear map.
-
-The naming is derived from `finsupp.dom_congr`, noting that here the permutation applies to the
-domain of the domain. -/
 @[simps apply]
-def dom_dom_congr [add_comm_monoid M₃] [semimodule R M₃]
-  (m : multilinear_map R (λ i : ι, M₂) M₃) (σ : equiv.perm ι) :
-  multilinear_map R (λ i : ι, M₂) M₃ :=
+private def dom_dom_congr_aux {ι₁ ι₂ : Type*} [decidable_eq ι₁] [decidable_eq ι₂]
+  [add_comm_monoid M₃] [semimodule R M₃]
+  (σ : ι₁ ≃ ι₂) (m : multilinear_map R (λ i : ι₁, M₂) M₃) : multilinear_map R (λ i : ι₂, M₂) M₃ :=
 { to_fun := λ v, m (λ i, v (σ i)),
   map_add' := λ v i a b, by { simp_rw function.update_apply_equiv_apply v, rw m.map_add, },
   map_smul' := λ v i a b, by { simp_rw function.update_apply_equiv_apply v, rw m.map_smul, }, }
 
+/-- Transfer the equivalence between argument indices to an equivalence between maps
+The naming is derived from `finsupp.dom_congr`, noting that here the permutation applies to the
+domain of the domain. -/
+@[simps apply symm_apply]
```

Comment:
Will this generate a simp-lemma in terms of `dom_dom_congr_aux` (that will then also be simped away)?
If so, I prefer to have a direct simp-lemma that by-passes `dom_dom_congr_aux` (which is an implementation detail).

Reply:
I discovered that this does not work with dot notation anyway - so I've made `dom_dom_congr_aux` part of the public API, and renamed both definitions